### PR TITLE
Added Terraform backend implementation for OCI Object Storage

### PR DESCRIPTION
**Pull request by ravinitp**
# Terraform backend implementation for OCI Object Storage
A new Terraform backend utilising Oracle Cloud Infrastructure (OCI) Object Storage as the storage backend. Leveraging OCI Object Storage provides a scalable and cost-effective alternative to traditional backends, offering a robust solution for state file storage and management.
[RFC#34465](https://github.com/hashicorp/terraform/issues/34465)
[RFC#32634](https://github.com/hashicorp/terraform/issues/32634)

## Added 
Terraform backend implementation for OCI Object Storage

## Target Release

1.12.x

## CHANGELOG entry

- [x] This change is user-facing and I added a changelog entry.
- [ ] This change is not user-facing.

### CODE REVIEW

**Comment by ravinitp** [comment]
| Test Case               | Code                                                                                                                         | comand                                          | Result  | comment                                                                                   |
|-------------------------|----------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------|---------------|-------------------------------------------------------------------------------------------|
| only required attribute |  terraform {<br>      backend "oci" {<br>        bucket    = "state-file-bucket"<br>        namespace = "my-namespace"<br>      }  | ./terraform apply                               | Success |                                                                                           |
| optional attribute with basic auth | terraform { backend "oci" {<be> bucket = "terraform-state" <br> key = "etf.json" <br> namespace = "namespace" <br> tenancy_ocid = "ocid1.tenancy.oc1..xxxxx" <br> user_ocid = "ocid1.user.oc1..xxxxx"  <br> fingerprint = "bd:9e:49:9c:39:1e:40:73:0c:91"  <br> private_key_path = "~/ssh/private_key_25-04-08T10_29_58.508Z.pem" <br> region = "us-ashburn-1" <br> } } | ./terraform apply | success |                                                                 |
| new workspace env       |                 -                                                                                                    | ./terraform workspace new dev <br> ./terraform apply | success | After successful command able to see file tf-state-env/dev/terraform.tfstate and .md file |
| list workspace env      |                 -                                                                                                    | ./terraform workspace list                      | success | Able to see list of workspace env default *dev                                            |
|delete workspace | -                                                                      |  ./terraform workspace select default  <br> ./terraform workspace delete dev                            | success | state file in dev env got deleted

File: internal/backend/remote-state/oci/auth.go [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+		if readFileErr != nil {
+			return nil, fmt.Errorf("can not read private key from: '%s', Error: %q", p.privateKeyPath, readFileErr)
+		}
+		return common.PrivateKeyFromBytesWithPassword(pemFileContent, []byte(p.privateKeyPassword))
+	}
+
+	return nil, fmt.Errorf("can not get private_key or private_key_path from Terraform configuration")
+}
+
+func (p ociAuthConfigProvider) getConfigProviders() ([]common.ConfigurationProvider, error) {
```
Comment (pvkrishnachaitanya) [inline internal/backend/remote-state/oci/auth.go:149]: Can we reuse this with getConfigProviders of provider.go? or refactor to avoid code duplication?

File: internal/backend/remote-state/oci/auth.go [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+			}).DialContext,
+			TLSHandshakeTimeout: getDurationFromEnvVar(TLSHandshakeTimeout, DefaultTLSHandshakeTimeout),
+			TLSClientConfig:     &tls.Config{MinVersion: tls.VersionTLS12},
+			Proxy:               http.ProxyFromEnvironment,
+		},
+	}
+	return
+}
+
+func getCertificateFileBytes(certificateFileFullPath string) (pemRaw []byte, err error) {
```
Comment (pvkrishnachaitanya) [inline internal/backend/remote-state/oci/auth.go:321]: Any reason, we created a copy of this? There is a similar method in helpers.go?

File: internal/backend/remote-state/oci/multipart_upload.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+func SplitSizeToOffsetsAndLimits(size int64) ([]int64, []int64, error) {
+	partSize := DefaultFilePartSize
+	totalParts := (size + partSize - 1) / partSize
+	if totalParts > MaxCount {
+		return nil, nil, fmt.Errorf("file exceeds maximum part count")
+	}
+	offsets, limits := make([]int64, totalParts), make([]int64, totalParts)
+	for i := range offsets {
+		offsets[i] = int64(i) * partSize
+		limits[i] = partSize
```
Comment (pvkrishnachaitanya) [inline internal/backend/remote-state/oci/multipart_upload.go:190]: Do we need array for limits as it is the same value?
Can we just return offsets, partSize, nil from this method
and use partSize in the caller like
end := start + partSize

File: internal/backend/remote-state/oci/auth.go (hunk 1 above)
Comment (ravinitp) [reply internal/backend/remote-state/oci/auth.go:149]: Hi @pvkrishnachaitanya ,
We can't import internal package of https://github.com/oracle/terraform-provider-oci/blob/master/internal/provider/provider.go#L291
Also data source of both function is different.

File: internal/backend/remote-state/oci/auth.go (hunk 2 above)
Comment (ravinitp) [reply internal/backend/remote-state/oci/auth.go:321]: same reason.
use of internal package not allowed across the module.

File: internal/backend/remote-state/oci/multipart_upload.go (hunk 3 above)
Comment (ravinitp) [reply internal/backend/remote-state/oci/multipart_upload.go]: Done

**Review by pvkrishnachaitanya — COMMENTED** [review]
Reviewed & Approved

**Comment by SarahFrench** [comment]
Hi @ravinitp, thanks for your PR!

Could you please let us know when this PR is ready for review, and mark the PR as draft until then? Force pushes to this branch make it hard for us to identify what code has changed since the last time we took a look.

**Comment by ravinitp** [comment]
Hi @SarahFrench 
The PR is now ready for review. I've squashed all the commits into a single one for clarity. No further changes will be pushed.
I'll rebase once the PR is approved. Please proceed with the review.

File: website/docs/language/backend/oci.mdx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    key               = "path/to/my/key"
+    workspace_key_prefix = "envs/"
+    kms_key_id        = "ocid1.key.oc1.iad.xxxxxxxxxxxxxx"
+    auth              = "APIKey"
+    config_file_profile = "DEFAULT"
+  }
+}
+```
+
+This assumes we have a bucket created called `mybucket`. The Terraform state is written to the key `path/to/my/key` and Terraform state md5 is written to the key `path/to/my/key.md5`.
```
Comment (SarahFrench) [inline website/docs/language/backend/oci.mdx:37]: I'm not familiar with OCI Object Storage- why does a file's MD5 need to be written to a paired file here?

**Review by SarahFrench — COMMENTED** [review]
👋🏻  Hi again, I've looked through this PR with a colleague and here's our initial feedback.

Firstly, could you please add some automated test coverage to this code? I see that [you've identified different scenarios for manual testing](https://github.com/hashicorp/terraform/pull/36872#issuecomment-2792667326), and this could be a good basis for starting to write automated tests. I'll include more info about testing below.

Similarly, I saw that there's auth-related code copied from `terraform-provider-oci` - are there any automated tests for that code that can also be copied here? 

Please see other review comments for more feedback and questions, thanks!

---

## Testing

### Scenarios

Ideally automated tests would cover these scenarios:
* Testing how valid and invalid configuration values are handled by validation logic.
* Testing creation of a new workspace, then listing all available workspaces, and finally deleting the non-default workspace.
* Testing that you can write and read state to an OCI Object
    * Edge cases - make sure to test single-part upload/multi-part upload/uploading large state files
* Testing that when you have two instances of the backend, if one of them locks the state then the other backend instance cannot acquire the lock.


### Examples to reference when writing your tests

The S3 and kubernetes backends' tests would be a good reference when adding tests, examples:
* s3 : [TestBackendConfig_PrepareConfigValidation](https://github.com/hashicorp/terraform/blob/8d2dffedb36a4cf34dc41828e24d1a2833c9c1f5/internal/backend/remote-state/s3/backend_test.go#L1199C6-L1199C47) shows how you could test how good/bad configuration values are handled by your backend. This doesn't require interaction with a real bucket.
* s3 : [TestBackendLockFileWithPrefix](https://github.com/hashicorp/terraform/blob/8d2dffedb36a4cf34dc41828e24d1a2833c9c1f5/internal/backend/remote-state/s3/backend_test.go#L2825C6-L2825C35) shows a test that actually provisions and uses a real s3 bucket during the test. You would need to create tests like this to check things like locking, returning an accurate list of workspaces etc.
* kubernetes : [TestBackendLocks](https://github.com/hashicorp/terraform/blob/8d2dffedb36a4cf34dc41828e24d1a2833c9c1f5/internal/backend/remote-state/kubernetes/backend_test.go#L61) - this uses some [test helpers](https://github.com/hashicorp/terraform/blob/main/internal/backend/testing.go#L272) to check what happens if two backend instances try to acquire the lock on the same state. 

Note that these tests would need to provision OCI object storage buckets to be used by the backend, and clean them up at the end of the test. This would require supplying the test with credentials. The s3 backend has a pattern of [asserting those ENVs are set before running a given test](https://github.com/hashicorp/terraform/blob/8d2dffedb36a4cf34dc41828e24d1a2833c9c1f5/internal/backend/remote-state/s3/backend_test.go#L2826).

**Comment by SarahFrench** [comment]
Also, for ease of review, please don't squash any commits or force-push in response to the review feedback - we'll squash merge this PR in the end anyway

File: internal/backend/remote-state/oci/util.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+func getEnvSettingWithBlankDefault(s string) string {
+	return getEnvSettingWithDefault(s, "")
+}
+
+func getEnvSettingWithDefault(s string, dv string) string {
+	v := os.Getenv(TfEnvPrefix + s)
+	if v != "" {
+		return v
+	}
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply internal/backend/remote-state/oci/util.go:27]: Thanks for the feedback!

We've updated the environment variable resolution logic to accommodate both backend-specific configuration and users who wish to reuse their existing Terraform input variable setup (e.g., for shared credentials). The resolution now follows this order of precedence:

1. `TF_BACKEND_OCI_` – backend-specific, highest priority to avoid naming collisions

2. `TF_VAR_` – supports users managing both infrastructure and backend with the same credentials

3. `OCI_` – aligns with OCI SDK conventions

No prefix – fallback/default option

This setup offers flexibility while still respecting clear scoping. Let us know if this looks good or if you have any additional suggestions!

File: go.mod [hunk 6]
```diff
@@ -53,13 +53,16 @@ require (
 	github.com/mitchellh/colorstring v0.0.0-20190213212951-d06e56a500db
 	github.com/mitchellh/go-homedir v1.1.0
 	github.com/mitchellh/go-linereader v0.0.0-20190213213312-1b945b3263eb
+	github.com/mitchellh/go-testing-interface v1.14.1
 	github.com/mitchellh/go-wordwrap v1.0.1
 	github.com/mitchellh/gox v1.0.1
 	github.com/nishanths/exhaustive v0.12.0
+	github.com/oracle/oci-go-sdk/v65 v65.88.1
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply go.mod:60]: Done

[the remark this answers is not in the document]
Comment (SarahFrench) [reply go.mod:60]: Please push those changes to this PR

File: internal/backend/remote-state/oci/backend.go [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+	namespace            string
+	workspaceKeyPrefix   string
+	kmsKeyID             string
+	SSECustomerKey       string
+	SSECustomerKeySHA256 string
+	SSECustomerAlgorithm string
+	client               *RemoteClient
+}
+
+func (b *Backend) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply internal/backend/remote-state/oci/backend.go:135]: Done

File: website/docs/language/backend/oci.mdx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+}
+```
+
+## Configuration
+
+The oci backend requires the configuration of the OCI namespace and Object Storage bucket where the Terraform state file will be stored.
+### Credentials and Shared Configuration
+
+> ⚠️ **Warning:**
+> We strongly recommend using **environment variables** to supply credentials and other sensitive data.
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply website/docs/language/backend/oci.mdx:144]: Done

File: internal/backend/remote-state/oci/backend.go [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+		return diags
+	}
+
+	if bucketVal, ok := getBackendAttr(obj, BucketAttrName); ok {
+		b.bucket = bucketVal.AsString()
+	}
+	if namespaceVal, ok := getBackendAttr(obj, NamespaceAttrName); ok {
+		b.namespace = namespaceVal.AsString()
+	} else {
+		diags.Append(tfdiags.AttributeValue(tfdiags.Error, "Missing Required Attribute", "Bucket name cannot be null", cty.GetAttrPath("namespace")))
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply internal/backend/remote-state/oci/backend.go]: Done

File: website/docs/language/backend/oci.mdx (hunk 4 above)
Comment (radeksimko) [reply website/docs/language/backend/oci.mdx:37]: @ravinitp I believe the question Sarah asked above is less about the MD5 hash in general but more about why does it need to be stored in a separate file.

What the S3 backend does today with MD5 hashing has different motivations, they are mostly explained in [the commit which added the relevant logic](https://github.com/hashicorp/terraform/commit/0022d224e8e6f040671f8c8d159fd4963b6838cf). One side effect is that the hash is stored in DynamoDB rather than S3.

Oracle's Object Storage - at least according to the [docs](https://docs.oracle.com/en-us/iaas/Content/Object/Concepts/objectstorageoverview.htm) - does not have the same characteristics that S3 had back in 2017 - i.e. Oracle's object storage has "strong consistency". Even if it was eventually consistent you if you were facing the same challenge, storing the hash in the same storage with the same consistency guarantees does not solve any consistency related challenges.

It is certainly a good practice to calculate checksums and compare those as part of every operation to ensure data integrity. It should not however require storing that hash as a separate file.

The Oracle Object Storage calculates and returns the hash in response headers, and so considering the consistency guarantees that should be sufficient to enable you to do the integrity checks mentioned.

For example, the [`PutObject` method](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/PutObject) accepts `Content-MD5` request header, which enables comparison on the server side and also returns `opc-content-md5` response header, which enables you to also do comparison in the code here. The SDK exposes that header as `OpcContentMd5` field on the [`PutObjectResponse`](https://pkg.go.dev/github.com/oracle/oci-go-sdk/v65@v65.88.1/objectstorage#PutObjectResponse) struct.

Similarly, the [`GetObject` method](https://docs.oracle.com/en-us/iaas/api/#/en/objectstorage/20160918/Object/GetObject) provides `content-md5` and `opc-multipart-md5`, either of which should enable comparison with locally computed MD5 hash.

As for tampering protection - I'm struggling to understand how the existing implementation provides that. Assuming the md5 file has the same ACLs - which it basically has to have - then any attacker can modify both the state file and that md5 file, making tampering invisible. I think that detection of tampering may need to be done on a different level, perhaps versioning, logging - I'm not familiar enough with OCI to suggest a specific solution but I assume the solution wouldn't involve any backend (client-side) logic.

File: internal/backend/remote-state/oci/util.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (radeksimko) [reply internal/backend/remote-state/oci/util.go]: `TF_` prefix is what we consider Terraform's/HashiCorp's own scope and so we'd prefer to not introduce any backend-specific logic that conflicts with that.

Assuming that your OCI provider is driven by the same SDK, which also supports `OCI_` I believe that the objective of allowing users to drive both the infrastructure management and backend with the same credentials can be achieved just with `OCI_` environment variables.

There should be no need for either `TF_VAR_` or `TF_BACKEND_` variables in your backend.

The only exception to the above scoping amongst backend implementations is the HTTP backend, which is maintained by our own internal team. None of the "third party" backends expect `TF_` prefixed environment variables.

File: website/docs/language/backend/oci.mdx (hunk 4 above)
Comment (ravinitp) [reply website/docs/language/backend/oci.mdx:37]: Thanks @radeksimko ,
I agree with your point. I’ll remove the additional MD5 file and instead compare the MD5 value returned in the upload response.
We're already passing the MD5 value as part of the upload request, so this should be sufficient for integrity checks.

File: internal/backend/remote-state/oci/util.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (ravinitp) [reply internal/backend/remote-state/oci/util.go]: Sure let me remove TF_VAR_ and TF_BACKEND_

File: website/docs/language/backend/oci.mdx (hunk 4 above)
Comment (ravinitp) [reply website/docs/language/backend/oci.mdx]: Done

File: go.mod (hunk 6 above)
[the remark this answers is not in the document]
Comment (ravinitp) [reply go.mod]: Done

**Comment by ravinitp** [comment]
@jbardin 
Added suggested tests
```bash
ravbhart-mac:oci ravbhart$ go test -v .
=== RUN   TestBackendBasic
    backend_test.go:26: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-67ff8613"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
--- PASS: TestBackendBasic (16.66s)
=== RUN   TestBackendLocked_FolceUnclock
    backend_test.go:44: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-67ff8624"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
    backend_test.go:49: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-67ff8624"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
    testing.go:303: TestBackend: testing state locking for *oci.Backend
    testing.go:303: TestBackend: testing state locking for *oci.Backend
--- PASS: TestBackendLocked_FolceUnclock (16.84s)
=== RUN   TestBackendBasic_multipart_Upload
    backend_test.go:70: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-67ff8635"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
--- PASS: TestBackendBasic_multipart_Upload (19.95s)
PASS
ok  	github.com/hashicorp/terraform/internal/backend/remote-state/oci
```

**Review by SarahFrench — COMMENTED** [review]
Thank you for adding some automated tests, here is some initial feedback and suggestions.

Posting the test output to the PR is really useful, so please continue to do so when you make changes to tests.

File: .changes/v1.13/NEW FEATURES-20250410-154805.yaml
Comment (SarahFrench) [inline .changes/v1.13/NEW FEATURES-20250410-154805.yaml:1]: Additionally, please move this changelog into the .changes/v1.12 folder if the intention is for this PR to be released in Terraform v1.12. I can then label this PR with `1.12-backport` to include this PR in the correct release branch once we merge.

File: internal/backend/remote-state/oci/backend_test.go [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+	defer deleteOCIBucket(ctx, t, b1.client.objectStorageClient, bucketName, *response.ETag, namespace)
+	// Test state locking and force-unlock
+	backend.TestBackendStateLocks(t, b1, b2)
+	backend.TestBackendStateForceUnlock(t, b1, b2)
+}
+func TestBackendBasic_multipart_Upload(t *testing.T) {
+	testACC(t)
+
+	ctx := context.Background()
+	DefaultFilePartSize = 100 //	100 Bytes
```
[the remark this answers is not in the document]
Comment (ravinitp) [reply internal/backend/remote-state/oci/backend_test.go:66]: Since another test case within the same test function is also adding a test resource, the resulting state file exceeds 100 bytes. To verify multipart upload, you can look for a request URL that includes the suffix &uploadPartNum=1.
[test.log](https://github.com/user-attachments/files/19777497/test.log)

File: .changes/v1.13/NEW FEATURES-20250410-154805.yaml
Comment (ravinitp) [reply .changes/v1.13/NEW FEATURES-20250410-154805.yaml:1]: Done

File: internal/backend/remote-state/oci/backend.go [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+
+var (
+	lockFileSuffix = ".lock"
+)
+
+func New() backend.Backend {
+	return &Backend{}
+}
+
+// New creates a new backend for OSS remote state.
```
Comment (radeksimko) [inline internal/backend/remote-state/oci/backend.go:25]:
```suggestion
// New creates a new backend for OCI remote state.
```

**Review by SarahFrench — COMMENTED** [review]
I noticed there is an issue with the [code consistency check](https://github.com/hashicorp/terraform/actions/runs/14493802159/job/40660962596?pr=36872) on this PR. Please run `make syncdeps` from the root of the repo and commit all the changes you see. It looks like those changes would include the `consul` and `oci` backend packages.

**Comment by osterman** [comment]
IMO, the term `oci` should be reserved for future support of the Open Container Initiative, since that’s what OCI commonly refers to for most people outside the Oracle Cloud ecosystem. It would be a phenomenal backend to support (even supported by Oracle as OCIR), that would be vendor-agnostic. That’s why I was initially excited about this PR—until I realized it was for Oracle's S3 alternative object storage. Terraform should definitely support an OCI, just not this one.

To avoid confusion, I’d suggest using `oracle_oci` as the backend type. It clearly identifies this as Oracle’s cloud and leaves room for future Open Container support under the more widely recognized oci namespace.

**Comment by radeksimko** [comment]
@ravinitp Can you please run `make syncdeps`?

**Comment by ravinitp** [comment]
@radeksimko 
Screenshot 2025-04-17 at 6 55 09 PM

@SarahFrench This is Instanceprincipal auth log
[Uploading instancePrincipalAuth.log…]()

**Comment by radeksimko** [comment]
@ravinitp I originally thought the dependency change in the `consul` backend was unrelated to this PR but it is. I assume the `objx` library is a shared transitive dependency and as a result of OCI being added, it impacts consul's `go.sum`.

With that in mind, can you please run `make syncdeps` and commit and push that change to the `consul` backend?

**Comment by radeksimko** [comment]
@ravinitp As per CI failure - Can you also run the following command and commit & push the outcome, please?

```
go tool golang.org/x/tools/cmd/goimports -w ./internal/backend/remote-state/oci
```

**Comment by ravinitp** [comment]
Done goimports and sync deps
```bash
ravbhart-mac:oci ravbhart$ go test -v ./
=== RUN   TestBackendBasic
    backend_test.go:30: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-68012121"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
--- PASS: TestBackendBasic (15.57s)
=== RUN   TestBackendLocked_ForceUnlock
    backend_test.go:48: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-68012131"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
    backend_test.go:53: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-68012131"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
    testing.go:303: TestBackend: testing state locking for *oci.Backend
    backend_test.go:62: TestBackend: testing state locking for *oci.Backend
    testing.go:303: TestBackend: testing state locking for *oci.Backend
    backend_test.go:64: TestBackend: testing state locking for *oci.Backend
--- PASS: TestBackendLocked_ForceUnlock (29.58s)
=== RUN   TestBackendBasic_multipart_Upload
    backend_test.go:76: TestBackendConfig on *oci.Backend with configtesting.synthBody{Filename:"<TestWrapConfig>", Values:map[string]cty.Value{"bucket":cty.StringVal("terraform-remote-oci-test-6801214e"), "key":cty.StringVal("testState.json"), "namespace":cty.StringVal("idxvx0yktto5")}}
--- PASS: TestBackendBasic_multipart_Upload (19.15s)
=== RUN   TestOCIBackendConfig_PrepareConfigValidation
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/null_bucket
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/null_namespace
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/key_with_double_slash
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/workspace_key_prefix_with_leading_slash
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/encryption_key_conflict
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/Invalid_encryption_key_combination
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/private_key_and_private_key_path_conflict
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/invalid_auth_method
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/empty_bucket
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/empty_namespace
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/key_with_leading_slash
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/key_with_trailing_slash
=== RUN   TestOCIBackendConfig_PrepareConfigValidation/missing_region_for_InstancePrinciple_auth
--- PASS: TestOCIBackendConfig_PrepareConfigValidation (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/null_bucket (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/null_namespace (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/key_with_double_slash (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/workspace_key_prefix_with_leading_slash (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/encryption_key_conflict (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/Invalid_encryption_key_combination (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/private_key_and_private_key_path_conflict (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/invalid_auth_method (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/empty_bucket (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/empty_namespace (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/key_with_leading_slash (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/key_with_trailing_slash (0.00s)
    --- PASS: TestOCIBackendConfig_PrepareConfigValidation/missing_region_for_InstancePrinciple_auth (0.00s)
PASS
ok  	github.com/hashicorp/terraform/internal/backend/remote-state/oci	65.652s
ravbhart-mac:oci ravbhart$ 
```

File: internal/backend/remote-state/oci/backend_test.go [hunk 12]
```diff
@@ -0,0 +1,472 @@
+package oci
```
Comment (radeksimko) [inline internal/backend/remote-state/oci/backend_test.go:1]:
```suggestion
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: BUSL-1.1

package oci
```

This should address the remaining CI failure

**Comment by crw** [comment]
> IMO, the term `oci` should be reserved for future support of the Open Container Initiative, since that’s what OCI commonly refers to for most people outside the Oracle Cloud ecosystem. 

Thanks for raising this! Given that the provider is already named `oci` (https://registry.terraform.io/providers/oracle/oci/latest/docs), we will continue to use the un-prefixed nomenclature for this backend. Thanks again for bringing this to our attention.

**Review by radeksimko — APPROVED** [review]
